### PR TITLE
Automated Jazzy Enhancements

### PR DESCRIPTION
**Pull request by ShihabMehboob**
Enhancements to automatic generation of Jazzy docs, depending on the label that exists in the specified repo on GitHub. In reference to this issue: https://github.ibm.com/arf/planning-swift-solutions/issues/642

### CODE REVIEW

File: install-swift.sh
Comment (EnriqueL8) [inline install-swift.sh]: Did you rebase? This was @KyeMaloy97 earlier PR that got merged...

File: teet
Comment (tunniclm) [inline teet]: I don't think you meant to check this file in, right...?

File: build-package.sh
Comment (EnriqueL8) [inline build-package.sh]: Is this supposed to be commented out?

File: build-package.sh
Comment (EnriqueL8) [inline build-package.sh]: Maybe divide this line and add commenting to explain the `sed` and `grep` commands

File: build-package.sh
Comment (EnriqueL8) [inline build-package.sh]: Do we only generate docs in PRs? I thought we generated once the PR is merged into master then travis runs? Could you explain the flow?

File: jazzy.sh
Comment (EnriqueL8) [inline jazzy.sh]: Why do we need a `.jazzy.yaml` in the tests?

**Review by EnriqueL8 — CHANGES_REQUESTED** [review]

File: build-package.sh
Comment (EnriqueL8) [reply build-package.sh]: Link to github API doc?

File: build-package.sh
Comment (ShihabMehboob) [reply build-package.sh]: It was the old method of just invoking the jazzy script without checks, has now been removed.

File: build-package.sh
Comment (ShihabMehboob) [reply build-package.sh]: Has been commented now.

File: build-package.sh
Comment (ShihabMehboob) [reply build-package.sh]: It's only in PRs, where it checks for labels called 'jazzy-doc', any time Package Builder is run, it checks through to also see whether the tag exists on the respective PR and then generates docs.

File: jazzy.sh
Comment (ShihabMehboob) [reply jazzy.sh]: This was a previous test, has now been removed.

File: build-package.sh
Comment (EnriqueL8) [reply build-package.sh]: okay, so why do we check for `master` in jazzy.sh?

Comment (ShihabMehboob) [reply build-package.sh]: That's a valid point, that was left over from when it wasn't only done on PRs, will remove that.

Comment (EnriqueL8) [reply build-package.sh]: So we generate the docs in the PR and then we push to master the docs [here](https://github.com/ShihabMehboob/Package-Builder/blob/00c9816a67c7fc470ec085a4460914a4e38b79ef/jazzy.sh#L45)? before merging the  PR? I think the better flow would be this:
- Merge the PR
- Travis Runs on Master
- Generate the docs.
- Push to master

What do you think?

Comment (EnriqueL8) [reply build-package.sh]: Or we could create another PR, just for the docs? After the first PR has been pushed to master

**Comment by EnriqueL8** [comment]
After a chat on slack, we have agreed that the flow will be the following:
- Travis runs 
- If the label `jazzy-doc` exists, it runs jazzy on the current PR branch
- Then pushes those generated docs in a new commit which will skip travis to the current PR branch

**Note** If a new commit is added after the docs have been generated, they will get overridden.

**Review by EnriqueL8 — APPROVED** [review]
LGTM, if test pass

File: build-package.sh
Comment (djones6) [inline build-package.sh:172]:
```
macbook:~ djones6$ uname
Darwin
```
I think this doesn't work, because you need an uppercase `D` for `Darwin`.

Comment (EnriqueL8) [reply build-package.sh:172]: I agree doesn't work, just tested it in a script on mac... I missed it when reviewing... Actually the removed if statement showed a capital `Darwin`

**Pull request merged**
